### PR TITLE
Fixes to code quality issues

### PR DESCRIPTION
**Pull request by de-sh**
Hey @cortexproject, 

I recently ran [DeepSource](https://deepsource.io) on my fork of this repo and found some [interesting code quality issues](https://deepsource.io/gh/de-sh/cortex) that I decided to fix, because of which I am opening this PR. DeepSource is a code review automation tool that detects code quality problems in contributions using static analysis and helps developers automatically fix some of them, saving a ton of time during code review.

The fixes made in this PR are:
- Refactor use of switch with single case as if-else
- Return nil as intended by if condition
- Omit redundant control flow statement
- Simplify returning boolean expression
- Added DeepSource config .toml

---

It would be great if you consider using DeepSource for running continuous analysis on the repos. It's free for open-source and detects 215+ issues in Go. All you need to do is follow these simple steps:

1. Merge this PR.

2. [Sign up](https://deepsource.io/signup/) on DeepSource with your GitHub account and grant access to this repo.

3. Activate analysis on this repo.

4. Done! 🎊

Do let me know if you face any issues, I shall be right here to provide any assistance possible!

### CODE REVIEW

File: pkg/chunk/encoding/varbit_helpers.go [hunk 1]
```diff
@@ -71,8 +71,5 @@ func isSignedIntN(i int64, n byte) bool {
 		return false
 	}
 	lower := upper - (1 << n)
-	if i < lower {
-		return false
-	}
-	return true
+	return i >= lower
```
Comment (bboreham) [inline pkg/chunk/encoding/varbit_helpers.go:74]: This file is copied from an older version of Prometheus; we do not modify such files unless critically necessary.

**Review by bboreham — COMMENTED** [review]
I expect you are a bot, because I don't think a human would have gone to so much trouble for such trivial changes.

I noted one of them in more detail, but broadly they are all arguable and such noise distracts from more important work.

**Comment by de-sh** [comment]
I am sorry if this PR has caused any trouble, I only intend to help opensource projects adopt [DeepSource](https://deepsource.io). And no, I am not a bot :)

File: pkg/chunk/encoding/varbit_helpers.go (hunk 1 above)
Comment (de-sh) [reply pkg/chunk/encoding/varbit_helpers.go:74]: I am sorry if this PR has caused any trouble, I only intend to help opensource projects adopt [DeepSource](https://deepsource.io). And no, I am not a bot :)

**Comment by pracucci** [comment]
I agree with @bboreham. I don't see this PR significantly valuable and looks more a marketing spam PR than a real valuable contribution. I would be in favour of closing it.

**Comment by pracucci** [comment]
_Closing_

**Pull request closed without merging**